subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c

```diff
@@ -512,6 +514,42 @@ static void commit_latest_cmd_id(void)
 	}
 }
 
+void nrf_provisioning_set_interval(int interval)
+{
+	LOG_DBG("Provisioning interval set to %d", interval);
+
+	if (interval != nxt_provisioning) {
+		char time_str[sizeof(STRINGIFY(2147483647))] = {0};
+		int ret;
+
+		nxt_provisioning = interval;
+		reschedule = true;
+
+		ret = k_mutex_lock(&np_mtx, K_NO_WAIT);
+		if (ret < 0) {
+			LOG_ERR("Unable to lock mutex, err: %d", ret);
+			return;
+		}
+		/* Let the provisioning thread run */
+		k_condvar_signal(&np_cond);
+		k_mutex_unlock(&np_mtx);
+
+		ret = snprintf(time_str, sizeof(time_str), "%d", interval);
+		if (ret < 0) {
+			LOG_ERR("Unable to convert interval to string");
+			return;
+		}
+
+		ret = settings_save_one(SETTINGS_STORAGE_PREFIX "/interval-sec",
+			time_str, strlen(time_str) + 1);
+		if (ret) {
```

Comment:
Do you know why we store the time as a string?

Settings storage should be able to store the `int` value.

Not asking to change here, but if you have time, that could be refactored.

Reply:
Probably because config commands contain key-value pairs in string format.